editor/src/components/canvas/remix/remix-utils.tsx

```diff
@@ -99,3 +128,397 @@ export function getRoutesFromRouteManifest(
+        : resolve(importOrigin, toImport)
+
+      forEachRight(alreadyResolved, (filepath) => resolvedFileNames.push(filepath))
+
+      const resolvedParseSuccess: Either<string, MapLike<any>> = attemptToResolveParsedComponents(
+        resolvedFromThisOrigin,
+        toImport,
+        innerProjectContents,
+        customRequire,
+        mutableContextRef,
+        topLevelComponentRendererComponents,
+        '/src/root.js',
+        {},
+        [],
+        [],
+        metadataContext,
+        NO_OP,
+        false,
+        alreadyResolved,
+        null,
+      )
+      return foldEither(
+        () => {
+          // We did not find a ParseSuccess, fallback to standard require Fn
+          return requireFn(importOrigin, toImport, false)
+        },
+        (scope) => {
+          // Return an artificial exports object that contains our ComponentRendererComponents
+          return scope
+        },
+        resolvedParseSuccess,
+      )
+    }
+    return createExecutionScope(
+      filename,
+      customRequire,
+      mutableContextRef,
+      topLevelComponentRendererComponents,
+      innerProjectContents,
+      filename,
+      fileBlobs,
+      hiddenInstances,
+      displayNoneInstances,
+      metadataContext,
+      NO_OP,
+      false,
+      null,
+    )
+  }
+
+  const executionScope = executionScopeCreator(projectContents)
+
+  const nameAndUid = getDefaultExportNameAndUidFromFile(projectContents, filename)
+
+  return {
+    executionScopeCreator: executionScopeCreator,
+    loader: executionScope.scope['loader'] as LoaderFunction | undefined,
+    action: executionScope.scope['action'] as ActionFunction | undefined,
+    rootComponentUid: nameAndUid?.uid ?? 'NO-ROOT',
+  }
+}
+export function getRoutesAndModulesFromManifest(
+  routeManifest: RouteManifestWithContents,
+  futureConfig: FutureConfig,
+  curriedRequireFn: CurriedUtopiaRequireFn,
+  curriedResolveFn: CurriedResolveFn,
+  metadataContext: UiJsxCanvasContextData,
+  projectContents: ProjectContentTreeRoot,
+  mutableContextRef: React.MutableRefObject<MutableUtopiaCtxRefData>,
+  topLevelComponentRendererComponents: React.MutableRefObject<
+    MapLike<MapLike<ComponentRendererComponent>>
+  >,
+  fileBlobs: CanvasBase64Blobs,
+  hiddenInstances: Array<ElementPath>,
+  displayNoneInstances: Array<ElementPath>,
+): GetRoutesAndModulesFromManifestResult | null {
+  const routeModuleCreators: RouteIdsToModuleCreators = {}
+  const routingTable: RemixStaticRoutingTable = {}
+
+  const rootJsFile = getProjectFileByFilePath(projectContents, `${ROOT_DIR}/root.js`)
+  if (rootJsFile == null || rootJsFile.type !== 'TEXT_FILE') {
+    return null
+  }
+
+  const rootJSRootElement = getDefaultExportedTopLevelElement(rootJsFile)
+  if (rootJSRootElement == null) {
+    return null
+  }
+
+  const routesByParentId = groupRoutesByParentId(routeManifest)
+  const routes: DataRouteObject[] = createClientRoutes(
+    routeManifest,
+    {},
```

Comment:
I think this is either wrong, or missing a TODO, because `createClientRoutes` takes a `routeModulesCache` for this param, which is used when trying to determine if a module needs re-evaluating, so since this is always empty here (and never mutated) the re-evaluation function will never be able to retrieve the module

Reply:
a "ref" (global var in this case) is passed down in https://github.com/concrete-utopia/utopia/pull/4162/commits/fbac022f2c91c1b9dcd98b423f0cc6712297d3a5, to be cleared up in a future PR